src/python/WMCore/ReqMgr/CherryPyThreads/StatusChangeTasks.py

```diff
@@ -64,6 +64,39 @@ def moveToArchivedForNoJobs(reqmgrSvc, wfStatusDict, logger):
                 for nextStatus in nextStatusList:
                     reqmgrSvc.updateRequestStatus(wf, nextStatus)
                     count += 1
+        logger.info("Total %s-archived: %d", status, count)
+
+    return
+
+
+def moveToArchived(wmstatsSvc, reqmgrSvc, archiveDelayHours, logger):
+    """
+    Handle normal-archived trantion.
+    By checking AgentJobInfo status we can check whether all the agent deleted the data.
+    TODO: still need to handle the case whether agent couch is not updated for a while and agent data gets deleted
+    """
+    
+    currentTime = int(time.time())
+    threshold = archiveDelayHours * 3600
+    count = 0
+    
+    statusTransition = {"announced": "normal-archived",
+                        "aborted-completed": "aborted-archived",
+                        "rejected": "rejected-archived"}
+    
+    inputConditon = {"RequestStatus": ["announced"], "AgentJobInfo": "NO_KEY"}
+    outputMask = ["AgentJobInfo", "RequestTransition"]
+    
+    for status, nextStatus in statusTransition.items():
+        inputConditon = {"RequestStatus": [status], "AgentJobInfo": "NO_KEY"}
+        for reqInfo in wmstatsSvc.getFilteredActiveData(inputConditon, outputMask):
+            if reqInfo["RequestName"] and (not reqInfo["RequestTransition"] or 
```

Comment:
It returns list of dictionary in this case also it should always contain RequestName key

>BTW, is there any case that RequestTransition would be empty or None? Or is that just an additional protection in case sh** happens?

There shouldn't be for all active request. (Old request might not contain this but it shouldn't Active anymore)